src/middlewared/middlewared/plugins/network.py

```diff
@@ -2338,9 +2338,15 @@ async def sync(self):
                     leases = await self.middleware.call('interface.dhclient_leases', interface)
                     reg_routers = re.search(r'option routers (.+);', leases or '')
                     if reg_routers:
-                        # Make sure to get first route only
                         ipv4_gateway = reg_routers.group(1).split(' ')[0]
-                        break
+                        if ipv4_gateway:
+                            # This means that the end-user hasn't specified
+                            # a default gateway in the webUI, however, we
+                            # have one from dhclient running on one of the
+                            # interfaces. In this scenario, there is nothing
+                            # to do so return.
+                            return
```

Comment:
We will not remove a gateway which is not in database but is configured, so this would be change of behavior ?

Reply:
Also ipv6 gateway would be ignored completely